File: ws/nacato/u2/ws16Libraries/StoreApp/src/ec/edu/espe/store/model/Product.java
package ec.edu.espe.store.model;
import utils.Tax;

/**
 *
 * @author Angie Nacato, Error 404, @ESPE
 */
public class Product {
    private int id;
    private String description;
    private float price;
    private float pvp;

    public Product(int id, String description, float price) {
        this.id = id;
        this.description = description;
        this.price = price;
        
        //TODO compute totaL price
        pvp = Tax.computeTotal(price,15F);
    }

    public Product(int id, String description, float price, float pvp) {
        this.id = id;
        this.description = description;
        this.price = price;
        this.pvp = pvp;
    }

    @Override
    public String toString() {
        return "Product\n{" + "id=" + id + ", description=" + description + ", price=" + price + ", pvp=" + pvp + '}';
    }
    
    /**
     * @return the id
     */
    public int getId() {
        return id;
    }

    /**
     * @param id the id to set
     */
    public void setId(int id) {
        this.id = id;
    }

    /**
     * @return the description
     */
    public String getDescription() {
        return description;
    }

    /**
     * @param description the description to set
     */
    public void setDescription(String description) {
        this.description = description;
    }

    /**
     * @return the price
     */
    public float getPrice() {
        return price;
    }

    /**
     * @param price the price to set
     */
    public void setPrice(float price) {
        this.price = price;
    }

    /**
     * @return the pvp
     */
    public float getPvp() {
        return pvp;
    }

    /**
     * @param pvp the pvp to set
     */
    public void setPvp(float pvp) {
        this.pvp = pvp;
    }
}

File: my-fresh-copy/hw/nacato/u2/hw09StoreApp/StoreApp.js/model/Product.js
const Tax = require('../libraries/Tax');

class Product {
    #id;
    #description;
    #price;
    #pvp;

    constructor(id, description, price, pvp = null) {
        this.#id = id;
        this.#description = description;
        this.#price = price;

        if (pvp === null) {
            // TODO compute total price
            this.#pvp = Tax.computeTotal(price, 15.0);
        } else {
            this.#pvp = pvp;
        }
    }

    get id() { return this.#id; }
    set id(id) { this.#id = id; }

    get description() { return this.#description; }
    set description(description) { this.#description = description; }

    get price() { return this.#price; }
    set price(price) { this.#price = price; }

    get pvp() { return this.#pvp; }
    set pvp(pvp) { this.#pvp = pvp; }

    toString() {
        return `Product\n{id=${this.#id}, description=${this.#description}, price=${this.#price}, pvp=${this.#pvp}}`;
    }
}

module.exports = Product;